test/runnable/test19122.d

```diff
@@ -0,0 +1,37 @@
+/*
+TEST_OUTPUT:
+---
+---
+*/
+
+struct HasDestructor
+{
+    ~this()
+    {
+        ++d;
+    }
+    this(this)
+    {
+        ++p;
+    }
+}
+
+struct S
+{
+    union
+    {
+        int i;
+        HasDestructor h;
+    }
+}
+
+int d, p;
+void main()
+{
+    {
+        S s;
+        s = s;
+        assert(p == 0); // Should fail.
+    }
+    assert(d == 0); // Should fail.
```

Comment:
Those comment are misleading. It should not fail, right ?

Reply:
The test suite is littered with comments like these because the test cases are copy-pasted as is. I’ve mentioned that several times but I’ve been overruled every time.

Reply:
I hope we use [`assert(0)`](https://github.com/dlang/dmd/pull/9909#issuecomment-496824330) instead and those lines just disappear.